game/video.cc

```diff
@@ -50,38 +50,38 @@ Video::Video(fs::path const& _videoFile, double videoGap): m_videoGap(videoGap),
 	// make ffmpeg here to get any exception in the current thread
 	auto ffmpeg = std::make_unique<VideoFFmpeg>(_videoFile, [this] (auto f) { push(std::move(f)); });
 	m_grabber = std::async(std::launch::async, [this, file = _videoFile, ffmpeg = std::move(ffmpeg)] {
-			int errors = 0;
-			std::unique_lock<std::mutex> l(m_mutex);
-			while (!m_quit) {
-				if (m_seek_asked) {
-					m_seek_asked = false;
-
-					auto seek_pos = m_readPosition;
-					// discard all outdated frame. To avoid races between clean and push, clean and push are done in this thread.
-					m_queue.clear();
-
-					UnlockGuard<decltype(l)> unlocked(l); // release lock during seek
-					ffmpeg->seek(seek_pos);
-					continue;
-				}
+		int errors = 0;
+		std::unique_lock<std::mutex> l(m_mutex);
+		while (!m_quit) {
+			if (m_seek_asked) {
+				m_seek_asked = false;
+
+				auto seek_pos = m_readPosition;
+				// discard all outdated frame. To avoid races between clean and push, clean and push are done in this thread.
+				m_queue.clear();
+
+				UnlockGuard<decltype(l)> unlocked(l); // release lock during seek
+				ffmpeg->seek(seek_pos);
+				continue;
+			}
 
-				try {
-					UnlockGuard<decltype(l)> unlocked(l); // release lock during possibly blocking ffmpeg stuff
-					ffmpeg->handleOneFrame();
-					errors = 0;
-				} catch (FFmpeg::Eof&) {
-					{
-						UnlockGuard<decltype(l)> unlocked(l); // release lock for possibly blocking calls
-						push(Bitmap()); // EOF marker
-						std::clog << "ffmpeg/debug: done loading " << file << std::endl;
-					}
-					m_cond.wait(l, [this]{ return m_quit || m_seek_asked; });
-				} catch (std::exception& e) {
+			try {
+				UnlockGuard<decltype(l)> unlocked(l); // release lock during possibly blocking ffmpeg stuff
+				ffmpeg->handleOneFrame();
+				errors = 0;
+			} catch (FFmpeg::Eof&) {
+				{
 					UnlockGuard<decltype(l)> unlocked(l); // release lock for possibly blocking calls
-					std::clog << "ffmpeg/error: " << file << ": " << e.what() << std::endl;
-					if (++errors > 2) { std::clog << "ffmpeg/error: FFMPEG terminating due to multiple errors" << std::endl; break; }
+					push(Bitmap()); // EOF marker
+					std::clog << "ffmpeg/debug: done loading " << file << std::endl;
```

Comment:
Due to concurrency bugs in logging implementation, things like that should be written as
`std::clog << "ffmpeg/debug: done loading " + file + "\n" << std::flush;`

... but it is noted that there are plenty of logging statements in Performous that still violate this, and sometimes messages do get garbled.

Reply:
I suggest a separate PR for fixing these, or to replace the logging system entirely, if anyone is up for that.

Reply:
ok, let's do that